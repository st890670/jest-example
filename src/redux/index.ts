import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import { combineReducers } from "redux";
import {
  useDispatch as useReduxDispatch,
  useSelector as useReduxSelector,
  TypedUseSelectorHook,
} from "react-redux";
import todoReduce from "./slice/todo";
import rootSaga from "./saga";

const sagaMiddleware = createSagaMiddleware();

const reducer = combineReducers({
    todo: todoReduce,
});

export const store = configureStore({
  reducer,
  middleware: [sagaMiddleware],
});

sagaMiddleware.run(rootSaga);

export const useDispatch = useReduxDispatch;
export const useSelector: TypedUseSelectorHook<ReturnType<typeof reducer>> =
  useReduxSelector;