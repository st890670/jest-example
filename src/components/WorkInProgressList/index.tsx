import { useMemo, useCallback } from "react";
import { useDispatch, useSelector } from "../../redux";
import { markAsFinished } from "../../redux/slice/todo";

function WorkInProgressList() {
  const dispatch = useDispatch();
  const workInProcessList = useSelector((state) => state.todo.workInProcess);

  const handleMarkAsFinished = useCallback(
    (uuid: string) => {
      dispatch(markAsFinished(uuid));
    },
    [dispatch]
  );

  const renderList = useMemo(
    () =>
      workInProcessList.map((item) => (
        <div
          key={item.id}
          className="cursor-default"
          onClick={() => handleMarkAsFinished(item.id)}
        >
          {item.text}
        </div>
      )),
    [workInProcessList, handleMarkAsFinished]
  );

  return (
    <div>
      <h2 className="font-bold text-lg cursor-default">
        Work In Progress List
      </h2>
      {renderList}
    </div>
  );
}

export default WorkInProgressList;
