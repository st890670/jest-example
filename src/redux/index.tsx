import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import { combineReducers } from "redux";
import {
  useDispatch as useReduxDispatch,
  useSelector as useReduxSelector,
  TypedUseSelectorHook,
} from "react-redux";
import todoReduce from "./slices/todo";
import rootSaga from "./sagas";
import React from "react";

const sagaMiddleware = createSagaMiddleware();

export const reducer = combineReducers({
  todo: todoReduce,
});

const store = configureStore({
  reducer,
  middleware: [sagaMiddleware],
});

sagaMiddleware.run(rootSaga);

export const ReduxProvider = ({ children }: { children: React.ReactNode }) => (
  <Provider store={store}>{children}</Provider>
);

export const useDispatch = useReduxDispatch;
export const useSelector: TypedUseSelectorHook<ReturnType<typeof reducer>> =
  useReduxSelector;
