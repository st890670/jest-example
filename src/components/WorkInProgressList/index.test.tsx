import { shallow } from "enzyme";
import * as redux from "../../redux";
import WorkInProgressList from "./index";

describe("Work In Progress List Unit Test", () => {
  const useDispatch = jest.spyOn(redux, "useDispatch");
  const useSelector = jest.spyOn(redux, "useSelector");

  beforeEach(() => {
    useDispatch.mockReturnValue(jest.fn());
    useSelector.mockImplementation(() => []).mockReturnValue([]);
  });

  it("should be same as previous snapshot", () => {
    const component = shallow(<WorkInProgressList />);
    expect(component).toMatchSnapshot();
  });
});

export {};
