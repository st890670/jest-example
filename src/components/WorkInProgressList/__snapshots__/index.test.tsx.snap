// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work In Progress List Unit Test should be same as previous snapshot 1`] = `
<div>
  <h2
    className="font-bold text-lg cursor-default"
  >
    Work In Progress List
  </h2>
</div>
`;
